### PR TITLE
PUBDEV-5872, PUBDEV-5871: Add missing objective functions to XGBoost Java Predictor

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostMojoModel.java [hunk 1]
```diff
@@ -22,6 +48,9 @@ public XGBoostMojoModel(String[] columns, String[][] domains, String responseCol
     super(columns, domains, responseColumn);
   }
 
+  // finalize MOJO initialization after all the fields are read
+  public void postReadInit() {}
```
Comment (Pscheidl) [inline h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostMojoModel.java]: Missing `@Override` ?

**Comment by michalkurka** [comment]
@Pscheidl I remove the snapshot reference before it goes to master

File: h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostMojoModel.java (hunk 1 above)
Comment (michalkurka) [reply h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostMojoModel.java]: This defines the method, there is no parent that would have this function.

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java [hunk 2]
```diff
@@ -299,6 +296,12 @@ public static BoosterParms createParams(XGBoostParameters p, int nClasses) {
     }
     params.put("nthread", nthread);
 
+    Log.info("XGBoost Parameters:");
+    for (Map.Entry<String,Object> s : params.entrySet()) {
+      Log.info(" " + s.getKey() + " = " + s.getValue());
+    }
+    Log.info("");
```
Comment (Pscheidl) [inline h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java]: Why ?

Comment (michalkurka) [reply h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java]: Yes, that was on purpose - it actually looks nice in the log ;)

File: h2o-genmodel-extensions/xgboost/src/test/java/hex/genmodel/algos/xgboost/XGBoostJavaMojoModelTest.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class XGBoostJavaMojoModelTest {
+
+  @Test
+  public void testObjFunction() { // make sure we have implementation for all supported obj functions
+    for (XGBoostMojoModel.ObjectiveType type : XGBoostMojoModel.ObjectiveType.values()) {
+      assertNotNull(XGBoostJavaMojoModel.getObjFunction(type.getId()));
```
[the remark this answers is not in the document]
Comment (michalkurka) [reply h2o-genmodel-extensions/xgboost/src/test/java/hex/genmodel/algos/xgboost/XGBoostJavaMojoModelTest.java]: Will add, thank you

**Pull request merged**